### PR TITLE
Remove obsolete reference to CValidationState from UpdateCoins.

### PR DESCRIPTION
**Pull request by ghost**
I prefer, but do not insist on this. Might want to refactor the remaining dummy CTxUndo txundo; at some other time.

### CODE REVIEW

File: src/main.h [hunk 1]
```diff
@@ -359,7 +359,7 @@ bool CheckInputs(const CTransaction& tx, CValidationState &state, const CCoinsVi
                  unsigned int flags, bool cacheStore, std::vector<CScriptCheck> *pvChecks = NULL);
 
 /** Apply the effects of this transaction on the UTXO set represented by view */
-void UpdateCoins(const CTransaction& tx, CValidationState &state, CCoinsViewCache &inputs, int nHeight);
+void UpdateCoins(const CTransaction& tx, CCoinsViewCache &inputs, int nHeight);
```
Comment (MarcoFalke) [inline src/main.h]: nit: If you do trivial fixes, please also do all other trivial fixes in the same line:

```diff
diff --git a/src/main.cpp b/src/main.cpp
index c5d2e23..7d5beab 100644
--- a/src/main.cpp
+++ b/src/main.cpp
@@ -1747 +1747 @@ void static InvalidBlockFound(CBlockIndex *pindex, const CValidationState &state
-void UpdateCoins(const CTransaction& tx, CCoinsViewCache &inputs, CTxUndo &txundo, int nHeight)
+void UpdateCoins(const CTransaction& tx, CCoinsViewCache& inputs, CTxUndo& txundo, int nHeight)
@@ -1773 +1773 @@ void UpdateCoins(const CTransaction& tx, CCoinsViewCache &inputs, CTxUndo &txund
-void UpdateCoins(const CTransaction& tx, CCoinsViewCache &inputs, int nHeight)
+void UpdateCoins(const CTransaction& tx, CCoinsViewCache& inputs, int nHeight)
diff --git a/src/main.h b/src/main.h
index 716378e..792747a 100644
--- a/src/main.h
+++ b/src/main.h
@@ -362 +362 @@ bool CheckInputs(const CTransaction& tx, CValidationState &state, const CCoinsVi
-void UpdateCoins(const CTransaction& tx, CCoinsViewCache &inputs, int nHeight);
+void UpdateCoins(const CTransaction& tx, CCoinsViewCache& inputs, int nHeight);

Comment (21E14) [inline src/main.h]: Upping the ante? Updated.

**Comment by dcousens** [comment]
utACK c8b9248

**Comment by sipa** [comment]
I like this. utACK c8b92486c4eca1287546b0683b6af3551db1fe67

**Comment by laanwj** [comment]
utACK c8b9248

**Pull request merged**